iothub/device/samples/PnpDeviceSamples/Thermostat/Program.cs

```diff
@@ -27,7 +28,21 @@ public class Program
         // DTDL interface used: https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/Thermostat.json
         private const string ModelId = "dtmi:com:example:Thermostat;1";
 
+        // This environment variable indicates if DPS or IoT Hub connection string will be used to provision the device.
+        // Expected values: (case-insensitive)
+        // "DPS" - The sample will use DPS to provision the device.
+        // "connectionString" - The sample will use IoT Hub connection string to provision the device.
+        private static readonly string s_deviceSecurityType = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_SECURITY_TYPE");
+
+        // Required if IOTHUB_DEVICE_SECURITY_TYPE is set to "connectionString".
         private static readonly string s_deviceConnectionString = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_CONNECTION_STRING");
```

Comment:
Any interest in using the CommandLineParser nuget, like I changed up recently in our samples repo? It isn't that big of a change. https://github.com/Azure-Samples/azure-iot-samples-csharp/commit/6097ef17fe76875c40241136cd45439ca187f21c

Reply:
Yes, I did want to do that, but maintaining similar usage experience across all languages was important here. So the decision was taken for all languages to use environment variables.

Reply:
We have argument handling all mixed in with the rest of the sample. This is bad form and makes it hard to read. Can you do that checking at program init?

Reply:
Here's a solution to have both https://github.com/commandlineparser/commandline/issues/677

Reply:
Perhaps not, it seems.

Reply:
I agree, it does make the sample cluttered. I'll decouple the args validation from the main sample.